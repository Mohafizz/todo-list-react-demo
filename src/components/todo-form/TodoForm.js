import React from "react";
import "./TodoForm.css";

const TodoForm = props => (
  <div>
    <form onSubmit={props.submitForm}>
      <label>
        <input
          id="form-input"
          type="text"
          value={props.value}
          onChange={props.changeForm}
        />
      </label>
      <input id="form-submit" type="submit" value="Submit" />
    </form>
  </div>
);

export default TodoForm;
