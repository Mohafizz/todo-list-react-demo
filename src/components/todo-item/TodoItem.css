#title {
  font-family: Georgia;
  font-size: x-large;
  background: gold;
}

.done {
  text-decoration: line-through;
}

.notDone {
  text-transform: uppercase;
}
