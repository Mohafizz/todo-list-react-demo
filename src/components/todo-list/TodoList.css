#todo-list {
  padding: 10px;
  max-width: 200px;
  margin: 0 auto;
  background-color: greenyellow;
  text-align: left;
}

#todo-title {
  background-color: aqua;
}
