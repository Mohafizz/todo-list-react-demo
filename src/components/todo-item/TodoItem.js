import React from "react";
import "./TodoItem.css";

const TodoItem = props => {
  const strikeThrough = props.todo.isCompleted ? "done" : "";
  return (
    <li id="todo-item" className={strikeThrough} onClick={props.done}>
      {props.todo.description}
    </li>
  );
};
export default TodoItem;
