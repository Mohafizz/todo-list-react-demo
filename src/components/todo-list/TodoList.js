import React, { Component } from "react";
import { todos } from "../../utils/seedData";
import "./TodoList.css";
import TodoItem from "../todo-item/TodoItem";
import TodoForm from "../todo-form/TodoForm";
import PropTypes from "prop-types";

class TodoList extends Component {
  constructor() {
    super();
    this.state = {
      todos: todos,
      newtodo: { description: "", isCompleted: false }
    };
  }

  render() {
    return (
      <div id="todo-list">
        <h1 id="todo-title">{this.props.title}</h1>
        {this.state.todos.map((todo, i) => {
          return (
            <TodoItem
              key={i}
              todo={todo}
              done={this.taskCompleted.bind(this, i)}
            />
          );
        })}
        <br />
        <TodoForm
          submitForm={this.handleSubmit.bind(this)}
          value={this.state.newtodo.description}
          changeForm={this.handleChange.bind(this)}
        />
      </div>
    );
  }

  taskCompleted(todoIndex) {
    const todosCopy = [...this.state.todos];
    const todoTobeUpdated = todosCopy[todoIndex];
    todoTobeUpdated["isCompleted"]
      ? (todoTobeUpdated["isCompleted"] = false)
      : (todoTobeUpdated["isCompleted"] = true);
    this.setState({ todos: todosCopy });
  }

  handleChange(event) {
    this.setState({
      newtodo: { description: event.target.value, isCompleted: false }
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const todosCopy = [...this.state.todos];
    todosCopy.push(this.state.newtodo);
    this.setState({
      todos: todosCopy,
      newtodo: { description: "", isCompleted: false }
    });
  }
}

TodoList.propTypes = {
  title: PropTypes.string.isRequired
};

TodoList.defaultProps = {
  title: "My todo list"
};
export default TodoList;
