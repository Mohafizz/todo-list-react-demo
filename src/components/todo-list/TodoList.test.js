import React, { Component } from "react";
import { shallow } from "enzyme";
import TodoList from "./TodoList";
import TodoItem from "../todo-item/TodoItem";
import TodoForm from "../todo-form/TodoForm";

describe("TodoList ", () => {
  it("should render TodoList properly", () => {
    const wrapper = shallow(<TodoList title="todos" />);
    expect(wrapper.find("h1").props().children).toBe("todos");
    expect(wrapper.find("#todo-list")).toHaveLength(1);
    expect(wrapper.find("h1#todo-title")).toHaveLength(1);
    expect(wrapper.find(TodoItem)).toHaveLength(wrapper.state().todos.length);
    expect(wrapper.find(TodoForm)).toHaveLength(1);
  });
});
