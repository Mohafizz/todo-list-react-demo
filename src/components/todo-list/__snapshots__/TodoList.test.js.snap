// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList  should match the snapshot 1`] = `
<div
  id="todo-list"
>
  <h1
    id="todo-title"
  >
    My todo list
  </h1>
  <li
    className=""
    id="todo-item"
    onClick={[Function]}
  >
    buy milk
  </li>
  <li
    className=""
    id="todo-item"
    onClick={[Function]}
  >
    drink juice
  </li>
  <li
    className=""
    id="todo-item"
    onClick={[Function]}
  >
    eat lunch
  </li>
  <li
    className=""
    id="todo-item"
    onClick={[Function]}
  >
    learn react
  </li>
  <li
    className=""
    id="todo-item"
    onClick={[Function]}
  >
    go home
  </li>
  <br />
  <div>
    <form
      onSubmit={[Function]}
    >
      <label>
        <input
          id="form-input"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
      <input
        id="form-submit"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
